### DISCUSSION TITLE
pls help - transparent command side effect

### DISCUSSION THREAD

**Discussion by ram-sh** [ImageMagick/ImageMagick#4248] [Digital Image Processing] [answered]
Hello all,

I have a large set of images with various degrees of a blurred circle centered on a grey (#7f7f7f) background. I am using magick to set that  the grey background pixel color to transparent using this command:

`magick convert 19.png -transparent "#7f7f7f" transparent.png`

the result seems good at first glance, but when I put the image back on top of a grey background (again - #7f7f7f), there seems to be a light contour line all around the "cutting" area made by the transparent command.

Please look at the attached images (same image in 2 zoom levels), On the right side is the original circle "blurring" into the #7f7f7f background color. On the left side is the transparent circle placed on top of a grey background (so the strange borders can be seen)

does anyone know why is it, and/or how do i get rid of it?

Many thanks in advance!
Best,
Ram

![rOAkG](https://user-images.githubusercontent.com/5938042/133891450-2f925987-845f-4473-af4d-89a8aab186d6.png)
![cS6x8](https://user-images.githubusercontent.com/5938042/133891452-df9d6d6d-0fd0-4f06-8b2e-43f668ecb592.png)

original image:
![19](https://user-images.githubusercontent.com/5938042/133891637-0685ad44-3691-49f2-a6ac-8c913925a769.png)

**Comment by snibgo** [comment]
What version of IM, on what platform (bash, Windows, etc)?

In your first image, a thin white line is visible at what might be the border of what was transparent.

Your command clearly did not make that image from your "original image". Please show all the commands that made the first image from the last.

**Comment by ram-sh** [comment]
Thank you snibgo for your reply!

I'm using windows 10,  mogrify ver 7.0.10.28 and ImageMagick  ver 7.0.10.28; the following is the full process of my current try:
original file (19.pgm) is PGM format and I convert it to PNG using mogrify:
` mogrify -format png *.pgm`

I then convert the output (19.png) to change all the #7f7f7f pixels to full transparent:
`magick convert 19.png -transparent "#7f7f7f" transparent.png`

pls find all images in the attached zip file.

once many thanks for your support!
[support.zip](https://github.com/ImageMagick/ImageMagick/files/7190257/support.zip)

**Comment by snibgo** [comment, marked as answer]
But your transparent.png is mostly transparent, and your first image has no transparency. How did you make that first image?

You say:
> ... but when I put the image back on top of a grey background (again - #7f7f7f), there seems to be a light contour line ...

Doing that the obvious way with IM:
```
magick transparent.png -background "#7f7f7f" -layers Flatten x.png
```
... does not create a light contour line. I would not expect it to. But you have clearly done some other processing to create your first image. And something in that processing has introduced the light contour line.

**Reply by ram-sh** [reply]
that's great news for us :) so the problem was not with the transparent image but with the image that "combined" the background gray image as the lower layer and the transparent image on top of it (it was done with PowerPoint.....)

thank you so very much for helping us, I appreciate it very much!
